src/Entities/Player.cpp

```diff
@@ -1046,12 +1046,21 @@ void cPlayer::Respawn(void)
 	m_LifetimeTotalXp = 0;
 	// ToDo: send score to client? How?
 
-	m_ClientHandle->SendRespawn(GetWorld()->GetDimension(), true);
+	m_ClientHandle->SendRespawn(dimOverworld, true);
 
 	// Extinguish the fire:
 	StopBurning();
 
-	TeleportToCoords(GetLastBedPos().x, GetLastBedPos().y, GetLastBedPos().z);
+	cWorld *world = GetWorld();
+	if (world->GetDimension() != dimOverworld)
+	{
+		world = cRoot::Get()->CreateAndInitializeWorld(world->GetLinkedOverworldName(), dimOverworld, world->GetLinkedOverworldName(), false);
```

Comment:
We should really move away from `CreateAndInitializeWorld` because it's really unsafe. The only reason it's still used is because there is no alternative. I'd say, if there is no proper world then just ignore the action and bail out.